File: 1922-count-good-numbers/1922-count-good-numbers.py
class Solution:
    def countGoodNumbers(self, n: int) -> int:
        mod = 1000000007
        if n % 2:
            return (pow(5,n//2+1,mod) * pow(4,n//2,mod)) % mod 
        return (pow(5,n//2,mod) * pow(4,n//2,mod)) % mod 

File: 1922-count-good-numbers/1922-count-good-numbers.cpp
class Solution {
public:
    long long power(long long x, long long n){
        int p = 1e9 + 7;
        x = x % p;
        if(x == 0)
            return 0;
        long long ans = 1;
        while(n){
            if(n & 1)
                ans = ( ans * x ) % p;
            x = ( x * x ) % p;
            n = n >> 1;
        }
        ans = ans % p;
        return ans;
    }
    int countGoodNumbers(long long n) {
        int p = 1e9 + 7;
        if(n % 2)
            return ( power(5,n/2 + 1) * power(4, n / 2) ) % p;
        return ( power(5,n/2) * power(4, n / 2) ) % p;
    }
};